kernel/aster-nix/src/fs/inode_handle/dyn_cap.rs

```diff
@@ -61,6 +64,101 @@ impl InodeHandle<Rights> {
         }
         self.0.readdir(visitor)
     }
+
+    pub fn position(&self) -> Result<i64> {
+        Ok(self.0.offset() as i64)
+    }
+
+    pub fn test_range_lock(&self, lock: &mut RangeLock) -> Result<()> {
+        let extension = match self.dentry().inode().extension() {
```

Comment:
If the `extension` method does not return an `Option`, the code can be simplified.